spring-boot-project/spring-boot-actuator/src/main/java/org/springframework/boot/actuate/quartz/QuartzEndpoint.java

```diff
@@ -0,0 +1,202 @@
+/*
+ * Copyright 2012-2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.boot.actuate.quartz;
+
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+import org.quartz.Job;
+import org.quartz.JobDetail;
+import org.quartz.JobKey;
+import org.quartz.Scheduler;
+import org.quartz.SchedulerException;
+import org.quartz.Trigger;
+import org.quartz.impl.matchers.GroupMatcher;
+
+import org.springframework.boot.actuate.endpoint.annotation.Endpoint;
+import org.springframework.boot.actuate.endpoint.annotation.ReadOperation;
+import org.springframework.boot.actuate.endpoint.annotation.Selector;
+import org.springframework.util.Assert;
+
+/**
+ * {@link Endpoint} to expose Quartz Scheduler info.
+ *
+ * @author Vedran Pavic
+ * @since 2.0.0
+ */
+@Endpoint(id = "quartz")
+public class QuartzEndpoint {
+
+	private final Scheduler scheduler;
+
+	public QuartzEndpoint(Scheduler scheduler) {
+		Assert.notNull(scheduler, "Scheduler must not be null");
+		this.scheduler = scheduler;
+	}
+
+	@ReadOperation
+	public Map<String, Object> quartzReport() {
+		Map<String, Object> result = new LinkedHashMap<>();
+		try {
+			for (String groupName : this.scheduler.getJobGroupNames()) {
+				List<String> jobs = this.scheduler
+						.getJobKeys(GroupMatcher.jobGroupEquals(groupName)).stream()
+						.map(JobKey::getName).collect(Collectors.toList());
+				result.put(groupName, jobs);
+			}
+		}
+		catch (SchedulerException ignored) {
+		}
+		return result;
+	}
+
+	@ReadOperation
+	public QuartzJob quartzJob(@Selector String groupName, @Selector String jobName) {
+		try {
+			JobKey jobKey = JobKey.jobKey(jobName, groupName);
+			JobDetail jobDetail = this.scheduler.getJobDetail(jobKey);
+			List<? extends Trigger> triggers = this.scheduler.getTriggersOfJob(jobKey);
+			return new QuartzJob(jobDetail, triggers);
+		}
+		catch (SchedulerException e) {
+			return null;
```

Comment:
Hey, I new here. So this is just a question why we didn't log what is the error?